src/dtemplate.d

```diff
@@ -1688,7 +1688,7 @@ public:
                 default:
                     goto Lnomatch;
                 }
-                ++argi;
+                assert(0);
```

Comment:
The easiest way to find out what caused the regression is to back all these changes out, then add them one by one until the regression happens.